src/segment_reader.rs

```diff
@@ -450,28 +461,38 @@ impl PrefetchingAsyncSegmentReader {
     fn issue_request_if_needed(&mut self) {
         if !self.end_of_segment && self.receiver.is_none() {
             let (sender, receiver) = oneshot::channel();
+            let (bg_sender, bg_receiver) = oneshot::channel();
             self.handle.spawn(PrefetchingAsyncSegmentReader::read_async(
                 self.reader.clone(),
                 sender,
+                bg_receiver,
                 self.offset + self.available() as i64,
                 self.read_size as i32,
             ));
             self.receiver = Some(receiver);
+            self.bg_job_handle = Some(bg_sender);
         }
     }
 
     // issues the read in the background
     async fn read_async(
         reader: Arc<Box<dyn AsyncSegmentReader>>,
         sender: oneshot::Sender<Result<SegmentReadCommand, ReaderError>>,
+        shutdown: oneshot::Receiver<bool>,
         offset: i64,
         length: i32,
     ) {
-        let result = reader.read(offset, length).await;
-        sender
-            .send(result)
-            .map_err(|e| warn!("failed to send reply back: {:?}", e))
-            .expect("send reply back");
+        tokio::select! {
+            result = reader.read(offset, length) => {
+                sender
+                .send(result)
+                .map_err(|e| warn!("failed to send reply back: {:?}", e))
```

Comment:
This error occurs regularly with pravega-video-server. It should not print the whole event and it should not panic. I had to apply the patch https://github.com/claudiofahey/pravega-client-rust/commit/70f6cf81e9b035fc8fe0ec338753ca001caebe86.